### DISCUSSION TITLE
TypeError: 'Network' object is not callable

### DISCUSSION THREAD

**Discussion by simonlindgren** [mapequation/infomap#105] [Q&A] [answered]
I am running infomap in two different python environments. Infomap is installed in the same way in both, and I run the same edgelist file in both. In one of the environments it works, and in the other I get:
```
infomapX.network().addLink(*e)
TypeError: 'Network' object is not callable
```
What might be the problem here?

**Comment by simonlindgren** [comment]
Thanks! Changing my code to `infomapX.network.addLink(*e)` solved it!

**Comment by danieledler** [comment]
Hi, since v1.0 we have added a more 'pythonic' api, where for example the network is accessed with a getter property instead of a function (`.network` instead of `.network()`). We have documented the new API [here](https://mapequation.github.io/infomap/).

The reason it works in one environment probably means it doesn't have the v1.0 version but still running the beta. We dropped support for python 2 in v1.0, are you maybe running python 2 on the working environment?

**Comment by danieledler** [comment]
Sorry for the out-of-sync new reply, I happened to press enter too early before. So please check the second part of the answer.

**Comment by simonlindgren** [comment]
I am running python 3, and it seems `1.0.0b57` where the old code still works, but `1.0.5` where change was needed.

**Comment by danieledler** [comment]
Ok, so in the old working environment, you are running python 3 but still `pip install --upgrade infomap` doesn't give you the `1.0.5` version? Is there an error in the upgrade?

**Comment by simonlindgren** [comment]
No, all is good. It was just me not having upgraded one of the environments.

**Comment by danieledler** [comment, marked as answer]
Ah, I see, good! Just to make it more clear from the updated reply above, we have the new python documentation here: https://mapequation.github.io/infomap/python/

You can also check the updated [examples](https://github.com/mapequation/infomap/tree/master/examples/python) in the repo.
